CMakeLists.txt

```diff
@@ -42,6 +42,10 @@ if(LLVM_FOUND)
     message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
     if(LLVM_LINK_LLVM_DYLIB)
         set(AnyDSL_LLVM_LINK_SHARED "USE_SHARED")
+    else()
+        if (BUILD_SHARED_LIBS)
+            message(SEND_ERROR "Cannot build thorin as a shared library with the current build of LLVM. Build LLVM with LLVM_LINK_LLVM_DYLIB or change BUILD_SHARED_LIBS to off.")
```

Comment:
Why not ?

Reply:
Due to the use of global data structures in LLVM, it is not possible to link against static LLVM libraries multiple times. This will build though, but end up causing runtime errors.
Hence, either your LLVM provides a shared library to link against or you have to build static libraries in the AnyDSL toolchain.

Reply:
This is fine for Thorin and should be only a warning if it affects downstream projects.

Reply:
Actually, it affects ALL downstream projects.
Therefore, we should discourage this build configuration.

Reply:
I think Artic itself works just fine with this config, because it doesn't link with LLVM

Reply:
Actually, it does link against LLVM. This is due to the poor include management in thorin.
`artic/src/main.cpp` includes `thorin/src/thorin/be/llvm/cpu.h` which results in a transitive link dependency.

However, the issue already arises from
```
LLVM
|   \
|   RV
|   /
thorin
```
and it gets worse when building the runtime with jit support enabled.